### DISCUSSION TITLE
Version-dependent RandomMotion behavior

### DISCUSSION THREAD

**Discussion by mariakawula** [TorchIO-project/torchio#901] [General]
Hi all,

using the latest torchio (0.18.78) I'm getting very strong and unrealistic deformation if compared to 0.18.46. 
I'm running the same snippet with the two aforementioned versions and getting totally different results:
`RandomMotion(include=['image', 'label'], p=1, degrees=(0, 10), translation=(0, 30))` 
I tested it multiple times so I'm certain, it's not due to the random character of the deformation:
left 0.18.46 vs. right 0.18.78
![Screenshot from 2022-06-13 17-55-28](https://user-images.githubusercontent.com/46919840/173397369-0227d45c-9d2a-4fa2-9c95-a146f7c8fc29.png)

Were there any changes related to RandomMotion? Cannot find any note on that.
The older version gives very nice artifacts that I can imagine occur due to 3 cm patient motion. The one on the right is way too much.

**Comment by fepegar** [comment]
Hi, @mariakawula. Can you try setting the PyTorch seed before calling the transform with both versions?

```python
import torch
torch.manual_seed(1234)
```

**Comment by romainVala** [comment]
with the same seeding in the two version you should find the same deformation

I am not surprise to see those kind of different severity for similar motion range, this is due to when motion changes occur in time. you can find the information with the times parameter in the history of the transformed subject

If motion occur at the beginning or at the end it well only perturbs high frequency of the image, whereas if it happen in the middle of the experiment you may induce large contrast change or object doubling

**Comment by mariakawula** [comment]
I set the manual seed as you both suggested and did reproduce deformations within one version. There is still the inter-version difference though.  I'm still trying to get to the history and find this 'time' parameter. I will let you know how it goes. 

Do you have other ideas on what can cause it? The newer version slows down the convergence during training and the model achieves only sub-optimal performance. Of course, I can come back to 0.18.46 but this bothers me and makes me wonder whether there were some changes done to the source code of torchio

**Comment by romainVala** [comment]
Thanks you for investigating, it is important to report if there are changes between both version (I do not see why ...).
it is important to check all random parameters (times translations and degree) to get them
you simply do something like 

```python

tmot = RandomMotion(include=['image', 'label'], p=1, degrees=(0, 10), translation=(0, 30))
suj_tranform = tmot(suj)
suj_transform.history

```

**Reply by mariakawula** [reply]
Yes. To be sure I'm not missing on something, I posted the snippet I'm using below

**Comment by mariakawula** [comment]
It's slightly more convoluted since I'm using Compose transforms and DataLoaders but the core is the same:
```
torch.manual_seed(1234)
tmot = Compose(
    [
        LoadImaged(keys=["image", "label"]),
        AddChanneld(keys=['image', 'label']),
        RandomMotion(include=['image', 'label'], p=1, degrees=(0, 10), translation=(0, 30)),
        ToTensord(keys=['image', 'label']),
    ]
)

train_ds = Dataset(data=train_files, transform=tmot)
training_data_loader = DataLoader(train_ds, batch_size=args.batch_size, shuffle=False,
                                  num_workers=4, pin_memory=torch.cuda.is_available())

for number, batch in enumerate(training_data_loader):
    #save some images 

```
I'm using a combination of monai, torch, and tochio funcions. Everything runs inside a monai 0.6.0 container.  In the meantime I'm uninstalling one torchio version and installing the other one

**Comment by fepegar** [comment]
See https://github.com/fepegar/torchio/issues/911. I introduced this bug in `v0.18.77`, apologies. It will be fixed in `v0.18.80`.
